packages/block-editor/src/hooks/test/generated-variation-class-name.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import {
+	registerBlockType,
+	unregisterBlockType,
+	registerBlockVariation,
+	unregisterBlockVariation,
+	getBlockType,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import '../generated-variation-class-name';
+
+describe( 'generated variation className', () => {
+	const defaultBlockSettings = {
+		attributes: {
+			fruit: {
+				type: 'string',
+				default: 'Bananas',
+				source: 'text',
+				selector: 'div',
+			},
+		},
+		save: ( { attributes } ) => (
+			// eslint-disable-next-line react/no-unknown-property
+			<div class={ attributes.className }>{ attributes.fruit }</div>
+		),
+		category: 'text',
+		title: 'block title',
+	};
+
+	const variationBlockSettings = {
+		attributes: {
+			fruit: {
+				type: 'string',
+				default: 'Bananas',
+				source: 'text',
+				selector: 'div',
+			},
+		},
+		isActive: ( { fruit } ) => fruit === 'Apples',
+	};
+
+	beforeAll( () => {
+		registerBlockType( 'core/test-block', {
+			...defaultBlockSettings,
+		} );
+
+		registerBlockVariation( 'core/test-block', {
+			name: 'variation',
+			title: 'Variation',
+			...variationBlockSettings,
+		} );
+	} );
+
+	afterAll( () => {
+		unregisterBlockType( 'core/test-block' );
+		unregisterBlockVariation( 'core/test-block', 'variation' );
+	} );
+
+	describe( 'addSaveProps', () => {
+		const addSaveProps = applyFilters.bind(
+			null,
+			'blocks.getSaveContent.extraProps'
+		);
+
+		it( 'should do nothing if the block settings do not define generated className support', () => {
+			const attributes = { className: 'foo' };
+			const extraProps = addSaveProps(
+				{},
+				{
+					...defaultBlockSettings,
+					supports: {
+						className: false,
+					},
+				},
+				attributes
+			);
+
+			expect( extraProps ).not.toHaveProperty( 'className' );
+		} );
+
+		it( 'should inject the generated className', () => {
+			const attributes = { fruit: 'Apples' };
+			const blockType = getBlockType( 'core/test-block' );
+			const variationBlockType = {
+				...blockType,
+				...variationBlockSettings,
+			};
+			const extraProps = addSaveProps(
+				{ className: 'foo' },
+				variationBlockType,
+				attributes
+			);
```

Comment:
I don't think we should merge `blockType` and `variationBlockSettings` into one object -- they're actually different in nature, see https://github.com/WordPress/gutenberg/pull/61864/files#r1655105267. AFAICT, the filter just needs the block type definition as its second argument:

```suggestion
			const extraProps = addSaveProps(
				{ className: 'foo' },
				blockType,
				attributes
			);
```

Reply:
Or even just

```suggestion
			const extraProps = addSaveProps(
				{ className: 'foo' },
				{ name: 'core/test-block', ...defaultBlockSettings },
				attributes
			);
```